src/hooks/listHooks.js

```diff
@@ -218,6 +218,13 @@ export const useRenderItem = (query, navigation, options = {}) => {
     default: {
       /* eslint-disable complexity */
       renderItem = ({ item, index, section, target }) => {
+        const bottomDivider =
+          query === QUERY_TYPES.PUBLIC_JSON_FILE
```

Comment:
there shouldn't be an exception for public json files. we want all lists behave the same, right?!

Reply:
I don't think there will be a problem. When I look at other textListItems I can see the Divider at the end of the list. For example:

<img width="437" alt="image" src="https://github.com/user-attachments/assets/4b3a73af-59f6-4bca-8f5f-3a2f3bda555f" />

Reply:
it is not a problem of working or not working. the code is bad with this specific condition on some query, if we want a divider in all cases

Reply:
I updated the code, now instead of the QUERY control, there is a `bottomDivider` control. And I added `calculateBottomDivider` function to prevent code repetition. This way I think I prevent repetitions in the code. Now I think we have a more understandable code.